### PR TITLE
Add documentation for debug error handling

### PR DESCRIPTION
**Pull request by Lunrtick**
This is a first attempt at documenting the error handling with DEBUG on or off, as discussed in https://github.com/vitalik/django-ninja/issues/619

### CODE REVIEW

**Comment by vitalik** [comment]
@Lunrtick thank you

I was actually thinking to add extra check - do not shallow exception if ATOMIC_REQUESTS is set (which I think is not for 95% users)

**Comment by Lunrtick** [comment]
That could definitely work. It would also mean that the user wouldn't get a stack trace back.

They could of course look in the terminal to see the error, so I suppose it doesn't matter too much either way.

Thinking about it now, maybe a Django middleware that comes as an optional extra with django-ninja would be the right way to go. You could keep lines 107 and 108 here and remove lines 109-112. 
https://github.com/vitalik/django-ninja/blob/e5e3a4c34afd136459dc00d6d2240604d3022bb5/ninja/errors.py#L104-L112

Instead, the logic could be
```python
if not settings.DEBUG: 
    raise exc  # let django deal with it
else:
    raise NinjaAPIException() from exc 
```
Or even just
```python
raise NinjaAPIException() from exc 
```
and then the middleware would be something simple like

```python
# catch NinjaAPIException
if settings.DEBUG:
    return 500, {"a nice": "stack trace"}
else:
    raise exc
```

This would mean that `ATOMIC_REQUESTS` should work as expected either way.

**Comment by thawancomt** [comment]
This behavior definitevily need changes, u know why? Data integrity, currently if we get an erro on the response cycle, da Database dont rollback cause no error get  the middleware...

so erro 500 and user saved? yes this is what is current happening.

My propose.


Current the default behavior is to set this handler:

that can be fixed by

```python

 def _default_exception( 
     request: HttpRequest, exc: Exception, api: "NinjaAPI" 
 ) -> HttpResponse: 
     if not settings.DEBUG: 
         raise exc  # let django deal with it 

     transaction.set_rollback(True) # Just add this line of code solve the entire problem
     logger.exception(exc) 
     tb = traceback.format_exc() 
     return HttpResponse(tb, status=500, content_type="text/plain") 

```

But the above code still getting happeing if user set a custom handler and dont handle the scenerio where a response cycle got an exception with is uncommon since return typacally isnt covered by transaction... maybe this last part didnt make sense, but u get the idea...

In a nutshell the final fix in on the mains,py. line 614

```python

    def on_exception(self, request: HttpRequest, exc: Exc[_E]) -> HttpResponse:
        handler = self._lookup_exception_handler(exc)
        transaction.set_rollback(True) # ensure in exception rollback is called
        if handler is None:
            raise exc
        return handler(request, exc)
```
